library/std/src/sys/switch/net.rs

```diff
@@ -204,7 +204,10 @@ impl Socket {
                 &mut addrlen,
             )
         })?;
-        Ok((n as usize, sockaddr_to_addr(&storage, addrlen as usize)?))
+        Ok((
+            n as usize,
+            sockaddr_to_addr(&storage, mem::size_of_val(&storage) as libc::socklen_t as usize)?,
```

Comment:
Should mention that the var `addrlen `change to 0 after `libc::recvfrom` execution and panic with the first assert in the `sockaddr_to_addr` function.
Don't exactly know why but i feel like this shouldn't be the case